### PR TITLE
Yinchen

### PR DESCRIPTION
**Pull request by enriqueChen**
add  configure  file loader.

### CODE REVIEW

File: mock_service.go [hunk 1]
```diff
@@ -31,33 +32,58 @@ type MockEndpoint struct {
...
+	if req.Method == http.MethodPost && req.URL.Path == m.mockCreatorEndpoint {
 		m.serveRegistrationHTTP(w, req)
 		return
 	}
 
 	m.serveMockHTTP(w, req)
 }
 
+func (m *MockService) PreloadEndpointsFromConf(confFilePath string) (error, bool){
+    if  len(m.mockCreatorEndpoint) < 1 {
```
Comment (wchan2) [inline mock_service.go]: Is the idea to block the possibility of the `mockEndpoint` being an empty string? If it is, it might be better to protect this in the `New` method instead. 

Its also a little clearer if we test for an empty string eg. `strings.Trim(m.mockRegistrationEndpoint) == ""` or something similar.

File: mock_service.go [hunk 2]
```diff
@@ -31,33 +32,58 @@ type MockEndpoint struct {
...
 }
 
-func New(createMockEndpoint string) *MockService {
+type MockEndpointSeries []MockEndpoint
+
+func New(mockCreatorEndpoint string) *MockService {
 	return &MockService{
-		createMockEndpoint: createMockEndpoint,
-		mockEndpoints:      map[string]map[string]MockEndpoint{},
+		mockCreatorEndpoint:  mockCreatorEndpoint,
```
Comment (wchan2) [inline mock_service.go]: What do you think about the name `mockRegistrationEndpoint`?

File: mock_service.go [hunk 3]
```diff
@@ -31,33 +32,58 @@ type MockEndpoint struct {
...
 	}
 
 	m.serveMockHTTP(w, req)
 }
 
+func (m *MockService) PreloadEndpointsFromConf(confFilePath string) (error, bool){
+    if  len(m.mockCreatorEndpoint) < 1 {
+        return  ErrCreatorNotExist, false
+    }
+    fi, err := os.Open(confFilePath)
```
Comment (wchan2) [inline mock_service.go]: Instead of the mock service opening the file, perhaps the constructor should take an argument of the pointer to some struct that defines the configuration file that is formatted in YAML, JSON, or so on. If its defined (non-nil) we can preload it, otherwise it'll create an empty `mockEndpoints` in the `*MockService*` returned.

That was the `io` could be unopinionated eg. the configuration could be read from somewhere else other than a file and passed in by the caller instead of the `mock_service` library defining where the configuration is coming from.

File: mock_service.go [hunk 4]
```diff
@@ -31,33 +32,58 @@ type MockEndpoint struct {
...
+        return err, false
+    }
+    for _, endpoint := range(series) {
+        m.CreateMockEndpoint(endpoint)
+    }
+    return nil, true
+}
+
 func (m *MockService) CreateMockEndpoint(endpoint MockEndpoint) {
+    if "POST" == endpoint.Method && m.mockCreatorEndpoint == endpoint.Endpoint {
```
Comment (wchan2) [inline mock_service.go]: I can see why this is added to protect this exported method from being called by a consumer who may not know that  registering on top of the `mockRegistrationEndpoint` will not be honored. I don't have a good way to solve this just yet. Perhaps we can un-export to protect it from those callers.

File: mock_service_test.go [hunk 5]
```diff
@@ -96,3 +96,30 @@ func TestServeMockHTTP(t *testing.T) {
 		t.Errorf(`Expected "%s" response body but got "%s"`, "hello world", testRecorder.Body.String())
 	}
 }
+
+func TestPreloadEndpointsFromConf(t *testing.T){
+    service := mock_service.New("/mocks")
+    service.PreloadEndpointsFromConf("test/test_mocker.json")
```
Comment (wchan2) [inline mock_service_test.go]: I suspect that you may need to commit the `test_mocker.json` file?

File: mock_service.go [hunk 6]
```diff
@@ -4,18 +4,19 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+    "os"
```
Comment (wchan2) [inline mock_service.go]: Lets use `go fmt -w` to format all the files 😄 .

File: mock_service.go (hunk 2 above)
Comment (enriqueChen) [inline mock_service.go]: I was, but a thought it is a little long... sone I use Creator. Although Registration is more consistent with the other context.

File: mock_service.go (hunk 6 above)
Comment (enriqueChen) [inline mock_service.go]: OK dude.

File: mock_service_test.go (hunk 5 above)
Comment (enriqueChen) [inline mock_service_test.go]: Yep. I missed the data file. Sorry dude.

File: mock_service.go (hunk 3 above)
Comment (enriqueChen) [inline mock_service.go]: You mean a default preload logic? I would implement it later.

Comment (wchan2) [inline mock_service.go]: I mean something like 

```go
type MockEndpoints struct {
    ...
}

func New(mockRegistrationEndpoint string, mockConf *MockConfig) *MockService {
    ms := &MockService{...}
    if mockConf != nil {
        // set up the ms.mockedEndpoints here, maybe by just calling `ms.preloadEndpoints(mockConf)` or something like that
    }
    return ms
}
```

File: mock_service_test.go (hunk 5 above)
Comment (wchan2) [inline mock_service_test.go]: No problemo!

**Comment by wchan2** [comment]
Alright I think I am going to merge this as the general idea is good but I will add some changes on top of it.
